## <center>Case Study on Testing of Hypothesis

In [1]:
import pandas as pd
Sd= pd.read_csv('Sales_add.csv')

In [2]:
# using Sd.info() to get an overall view about data set
Sd.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


### 1) Checking whether there is any increase in sales after stepping into digital marketing 

In [3]:
# Stating the hypothesis

H0= There is no significant increase in sales after stepping into digital marketing,µ1>=µ2

Ha=There is  significant increase in sales after stepping into digital marketing,µ1<µ2

In [4]:
# setting the significance level

Considering significance level(α)=0.05 or 5% 

In [5]:
# Determining p value using paired sample t-test (one tailed)

Checking whether there is any increase in sales ,so using the one-tailed t-test.
Need to compare means from same group at different times-before & after digital marketing- so using paired sample t-test.
Setting alternative='greater':Meaning the mean of the distribution underlying the first sample(after digital marketing)is greater than the mean of the distribution underlying the second sample(before digital marketing)


In [6]:
from scipy.stats import ttest_rel
ttest_rel(Sd['Sales_After_digital_add(in $)'],Sd['Sales_before_digital_add(in $)'],alternative='greater')

Ttest_relResult(statistic=12.09070525287017, pvalue=3.168333502287889e-11)

In [7]:
# comparing p value with significance level(0.05) 

p value: 3.168333502287889e-11 ≈ 3.16 x 10^-11 < 0.05

Rejecting the null hypothesis


In [8]:
# conclusion

Since p value is less than the selected significance level of 0.05 ,rejecting the null hypothesis which means that there is  significant increase in sales after stepping into digital marketing,µ1<µ2.

### 2) checking whether there is any dependency between the features “Region” and “Manager”

In [9]:
# Stating the hypothesis

H0= There is no dependency between the features "Region” and “Manager”

Ha= There is dependency between the features "Region” and “Manager”

In [10]:
# setting the significance level

Considering significance level(α)=0.05 or 5% 

In [11]:
# Determining p value using Chi-square test (test for independence)

In [12]:
from scipy import stats
crosstab = pd.crosstab(Sd["Region"],Sd["Manager"])
crosstab

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [13]:
stats.chi2_contingency(crosstab)

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

The first value ( ≈ 3.05) is the Chi-square value, followed by the p-value (≈ 0.55), then comes the degrees of freedom (4), and lastly it outputs the expected frequencies as an array.

In [14]:
# comparing p value with significance level(0.05) 

p value: 0.5493991051158094 ≈ 0.55 > 0.05 

Failing to reject the null hypothesis

In [16]:
# Conclusion

Since p value is greater than the selected significance level of 0.05 ,failing to reject the null hypothesis which means that  there is no dependency between the features "Region” and “Manager” 